File: Samples/Secrets Manager/TypeScript/retrieve-secret/secret-stack.ts
import * as cdk from 'aws-cdk-lib';
import {CfnOutput, RemovalPolicy, SecretValue, Tags} from 'aws-cdk-lib';
import {Construct} from 'constructs';
import {Topic} from 'aws-cdk-lib/aws-sns';
import * as secretsmanager from "aws-cdk-lib/aws-secretsmanager";
import * as lambda_ from "aws-cdk-lib/aws-lambda";

export class SecretStack extends cdk.Stack {
    constructor(scope: Construct, id: string, props?: cdk.StackProps) {
        super(scope, id, props);

        const secret = new secretsmanager.Secret(this, 'Secret', {
            removalPolicy: RemovalPolicy.DESTROY,
            secretStringValue: SecretValue.unsafePlainText('for your eyes only')
        })

        let secret_value = secret.secretValue.unsafeUnwrap()


        new CfnOutput(this, 'SecretName',
            {
                value: secret.secretName,
                description: 'The name of the secret.'
            }
        )

        new CfnOutput(this, 'SecretArn',
            {
                value: secret.secretArn,
                description: 'The ARN of the secret in AWS Secrets Manager.'
            }
        )


        /*
        Let's print the secret value out to demonstrate how it can be passed to other places.
         */
        new CfnOutput(this, 'SecretValueToken',
            {
                value: secret_value,
                description: 'The secret. Note: It is still the unresolved token at this stage.'
            }
        )

        /*
        Passing the resolved secret value to any resource, just to see it resolving."
         */
        const topic = new Topic(this, 'Topic', {})
        Tags.of(topic).add('Secret', secret_value)

        const fn = new lambda_.Function(this, 'Dysfunction', {
            runtime: lambda_.Runtime.NODEJS_18_X,
            handler: 'index.handler',
            code: lambda_.Code.fromInline(`exports.handler = handler.toString() //`),
            environment: {
                SECRET_VALUE: secret_value
            },
        });

        // Remember to grant our Lambda handler permission to retrieve our secret from inside the function!
        secret.grantRead(fn)

    }
}

File: Motley/Security/security/secret_stack.py
from aws_cdk import (
    # Duration,
    Stack,
    aws_secretsmanager as secretsmanager, CfnOutput, )
from aws_cdk.aws_sns import Topic
from constructs import Construct


class SecretStack(Stack):

    def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
        super().__init__(scope, construct_id, **kwargs)

        secret = secretsmanager.Secret(
            self,
            "TopSecret",
            generate_secret_string=secretsmanager.SecretStringGenerator(
                secret_string_template="{}",
                generate_string_key="auth_token",
                password_length=48,
                exclude_characters="':^,()@/",
                exclude_punctuation=True,
            ),
        )

        # How to resolve SecretValue
        secret_value = str(secret.secret_value_from_json("auth_token").unsafe_unwrap())

        dummy = Topic(self, 'Dummy',
                      topic_name=secret_value
                      )
        CfnOutput(self, 'SecretValue',
                  description='Resolved secret value',
                  value=dummy.topic_name
                  )
